### PR TITLE
Triplanes

### PR DESCRIPTION
**Pull request by jake-austin**
Triplanes seem to be working fine (this is with ns-train tensorf on triplanes with no extra args)

### CODE REVIEW

**Comment by jake-austin** [comment]
Screenshot 2023-02-19 at 3 18 36 PM

File: nerfstudio/field_components/encodings.py [hunk 1]
```diff
@@ -474,6 +474,91 @@ def upsample_grid(self, resolution: int) -> None:
...
+        """Upsamples underlying feature grid
+
+        Args:
+            resolution: Target resolution.
+        """
+        plane_coef = F.interpolate(
+            self.plane_coef.data, size=(resolution, resolution), mode="bilinear", align_corners=True
+        )
+
+        # TODO(ethan): are these torch.nn.Parameters needed?
```
Comment (tancik) [inline nerfstudio/field_components/encodings.py:557]: Can we resolve this?

**Review by tancik — COMMENTED** [review]
Thanks for adding into the main repo!

File: nerfstudio/field_components/encodings.py [hunk 2]
```diff
@@ -474,6 +474,91 @@ def upsample_grid(self, resolution: int) -> None:
...
+
+    The use for this is in representing a tensor decomp of a 4D embedding tensor: (x, y, z, feature_size)
+
+    This will return a tensor of shape (bs:..., num_components)
+
+    Args:
+        resolution: Resolution of grid.
+        num_components: The number of scalar triplanes to use (ie: output feature size)
+        init_scale: The scale of the initial values of the planes
+        product: Whether to use the element-wise product of the planes or the sum
```
[the remark this answers is not in the document]
Comment (jake-austin) [reply nerfstudio/field_components/encodings.py:499]: Sure! I like your wording of it much better

File: nerfstudio/field_components/encodings.py (hunk 1 above)
Comment (jake-austin) [reply nerfstudio/field_components/encodings.py:557]:
```
Printing profiling stats, from longest to shortest duration in seconds
Trainer.train_iteration: 0.0282              
VanillaPipeline.get_train_loss_dict: 0.0249              
Traceback (most recent call last):
  File "/home/jake-austin/miniconda3/envs/nerfstudio/bin/ns-train", line 8, in <module>
    sys.exit(entrypoint())
  File "/home/jake-austin/nerfstudio/scripts/train.py", line 247, in entrypoint
    main(
  File "/home/jake-austin/nerfstudio/scripts/train.py", line 233, in main
    launch(
  File "/home/jake-austin/nerfstudio/scripts/train.py", line 172, in launch
    main_func(local_rank=0, world_size=world_size, config=config)
  File "/home/jake-austin/nerfstudio/scripts/train.py", line 87, in train_loop
    trainer.train()
  File "/home/jake-austin/nerfstudio/nerfstudio/engine/trainer.py", line 207, in train
    callback.run_callback_at_location(step, location=TrainingCallbackLocation.AFTER_TRAIN_ITERATION)
  File "/home/jake-austin/nerfstudio/nerfstudio/engine/callbacks.py", line 103, in run_callback_at_location
    self.run_callback(step=step)
  File "/home/jake-austin/nerfstudio/nerfstudio/engine/callbacks.py", line 93, in run_callback
    self.func(*self.args, **self.kwargs, step=step)
  File "/home/jake-austin/nerfstudio/nerfstudio/models/tensorf.py", line 128, in reinitialize_optimizer
    self.field.density_encoding.upsample_grid(resolution)
  File "/home/jake-austin/miniconda3/envs/nerfstudio/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/jake-austin/nerfstudio/nerfstudio/field_components/encodings.py", line 558, in upsample_grid
    self.plane_coef = plane_coef
  File "/home/jake-austin/miniconda3/envs/nerfstudio/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1290, in __setattr__
    raise TypeError("cannot assign '{}' as parameter '{}' "
TypeError: cannot assign 'torch.cuda.FloatTensor' as parameter 'plane_coef' (torch.nn.Parameter or None expected)
```

This is the error thrown if you don't use an nn.Parameter around the plane_coef. I can look into this further if you feel it's worth trying to debug and see if we can get around, but idk if it's super worth trying to debug. I can remove the TODO if you want.

Comment (tancik) [reply nerfstudio/field_components/encodings.py:557]: Yea, remove the TODO

**Review by tancik — APPROVED** [review]
LGTM

**Pull request merged**